### PR TITLE
Improve merge imported

### PR DESCRIPTION
**Pull request by RobertJoonas**
### Changes

* Refactor the `Plausible.Stats.Imported` module (mainly the `merge_imported` function)
* Add support for new properties in `merge_imported`
  * `os_version`
  * `browser_version`
  * `utm_source`
  * `referrer`

### Tests
- [x] Automated tests have been added

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/stats/imported.ex [hunk 1]
```diff
@@ -448,6 +249,130 @@ defmodule Plausible.Stats.Imported do
...
+    })
+  end
+
+  defp group_imported_by(q, dim)
+       when dim in [:utm_source, :utm_medium, :utm_campaign, :utm_term, :utm_content] do
+    q
+    |> group_by([i], field(i, ^dim))
+    |> where([i], fragment("not empty(?)", field(i, ^dim)))
+    |> select_merge([i], %{
+      ^dim => fragment("if(empty(?), ?, ?)", field(i, ^dim), @no_ref, field(i, ^dim))
```
Comment (zoldar) [inline lib/plausible/stats/imported.ex:266]: Given `where` condition earlier in the pipeline, isn't `if` expression here redundant?

**Review by zoldar — APPROVED** [review]

File: lib/plausible/stats/imported.ex [hunk 2]
```diff
@@ -448,6 +249,130 @@ defmodule Plausible.Stats.Imported do
...
+
+  defp group_imported_by(q, dim) when dim in [:entry_page, :exit_page] do
+    q
+    |> group_by([i], field(i, ^dim))
+    |> select_merge([i], %{^dim => field(i, ^dim)})
+  end
+
+  defp select_joined_dimension(q, :city) do
+    select_merge(q, [s, i], %{
+      city: fragment("coalesce(nullif(?, 0), ?)", i.city, s.city)
```
Comment (ruslandoga) [inline lib/plausible/stats/imported.ex:351]:
```suggestion
      city: fragment("greatest(?,?)", i.city, s.city)
```

File: lib/plausible/stats/imported.ex (hunk 1 above)
Comment (RobertJoonas) [reply lib/plausible/stats/imported.ex]: You're right! https://github.com/plausible/analytics/pull/4003/commits/2bd857a745a7993ea7a8974c0a8970f5c8fc473a

File: lib/plausible/stats/imported.ex (hunk 2 above)
Comment (RobertJoonas) [reply lib/plausible/stats/imported.ex]: Good catch, thanks! 👍

**Pull request merged**